source/engine/engine/NcEngineImpl.cpp

```diff
@@ -50,6 +51,14 @@ auto BuildModuleRegistry(nc::Registry* reg,
     moduleRegistry.Register(std::make_unique<nc::Random>());
     return moduleRegistry;
 }
+
+auto BuildRegistry(size_t maxTransforms) -> nc::Registry
+{
+    // Register components immediately so they're available to modules
+    auto registry = nc::Registry{maxTransforms};
+    nc::RegisterEngineComponents(registry);
```

Comment:
Not sure if I'm a big fan of this builder method for the following reasons:

Is the purpose of ```nc::RegisterEngineComponents``` being extracted out like this to allow users to create their own instances of ```nc::Registry``` that don't contain engine components via calling the constructor directly?

If so, wouldn't that flow be broken because we call:
```
RegisterComponentType<Tag>();
RegisterComponentType<Transform>();
RegisterComponentType<ecs::detail::FreeComponentGroup>();
``` 
directly in the constructor body?

If that isn't the purpose, and ```nc::registry``` is only supposed to be for engine components, I don't like extracting out like this because it creates a hidden coupling at creation time for the registry

Reply:
The types that are left in the constructor of registry are the ones required for the registry to work. The ones extracted are here are the types used in the modules that get built right after this in the NcEngine constructor. Since the factory for the modules is right above this, it made sense to put the registry setup for them next to it. What if I moved the Registry and ModuleRegistry factories to the same file? Like rename ComponentRegistration.h to something else, and put them both there?